### DISCUSSION TITLE
[firebase_messaging] How to disable the default firebase behavior of showing a notification when the app is in the background

### DISCUSSION THREAD

**Discussion by iamrohitsuthar** [firebase/flutterfire#7454] [General]
I want to show notification with custom action buttons on it. For that I am creating notifications locally when I get callback in ```onBackgroundMessage```. But from backend I'm receiving notification payload and because of that firebase sdk automatically shows one notification on system tray. So now I am getting two notifications one from firebase sdk and another is my custom one. 

I know that in order to disable firebase sdk notification we can pass only data payload but, the android and ios platforms do not guarantees to show data messages specially ios. So I want to know is there any other solution to achieve this.

**Comment by SantiagoCor** [comment]
I'm having a very similar problem. I can't use data-only payload because iOS doesn't show them
